ash/src/prelude.rs

```diff
@@ -64,6 +65,40 @@ where
     }
 }
 
+/// Calls `f` twice until it does not return [`vk::Result::ERROR_NOT_ENOUGH_SPACE_KHR`], ensuring all
+/// available binary data has been read into the vector.
+///
+/// The first call happens with a [`Vec`] of size `4096`.  If this is not adequate, `f` is supposed
+/// to return [`vk::Result::ERROR_NOT_ENOUGH_SPACE_KHR`] while also updating `count` to the desired
+/// number of elements, allowing us to try again.
+///
+/// This function is _not_ designed to be used with [`vk::Result::INCOMPLETE`], see
+/// [`read_into_uninitialized_vector()`] instead.
+///
+/// See for example [`vkGetPipelineBinaryDataKHR()`], where the new return code was first introduced.
+///
+/// [`vkGetPipelineBinaryDataKHR()`]: https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetPipelineBinaryDataKHR.html
+pub(crate) unsafe fn read_into_uninitialized_binary_vector(
+    mut f: impl FnMut(&mut usize, *mut ffi::c_void) -> vk::Result,
+) -> VkResult<Vec<u8>> {
+    let mut count = 4096;
+    let mut data = Vec::<u8>::with_capacity(count);
+    let mut err_code = f(&mut count, data.as_mut_ptr().cast());
+    if err_code == vk::Result::ERROR_NOT_ENOUGH_SPACE_KHR {
+        debug_assert!(
+            count > 4096,
+            "Implementation should have updated the value to be higher than the initial request"
+        );
+        err_code = f(&mut count, data.as_mut_ptr().cast());
```

Comment:
This seems to be missing the code to actually expand the `Vec`.

```suggestion
        data.reserve(count);
        err_code = f(&mut count, data.as_mut_ptr().cast());
```